### PR TITLE
refactor: improve ListPoolsByDenom filter logic

### PR DESCRIPTION
**Pull request by levisyin**
Refer to [PR #6766](https://github.com/osmosis-labs/osmosis/pull/6766), get pool denoms info by `k.GetTotalPoolLiquidity` will lead to:
1. fetching pool module once again(one more io op) which already known as `k.poolModules` range iterator
2. filtering denom by `coins.AmountOf(denom).GT(osmomath.ZeroInt())` is odd, although there is no pool with zero amount token

## Testing and Verifying
This change is already covered by existing tests, such as *TestListPoolsByDenom*.

### CODE REVIEW

File: x/poolmanager/router.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"errors"
 	"fmt"
 	"math/big"
+	"slices"
```
Comment (mattverse) [inline x/poolmanager/router.go:7]: Seems like CI is not happy with the new package, can you use Contains that we have made instead? 
https://github.com/osmosis-labs/osmosis/blob/main/osmoutils/slice_helper.go#L98

**Review by mattverse — COMMENTED** [review]
Good catch once again! Thanks for this PR :) Let's merge this as soon as we make CI happy (slice package replacement)

File: x/poolmanager/router.go (hunk 1 above)
Comment (levisyin) [reply x/poolmanager/router.go]: Hi @mattverse , I have replaced it now

**Comment by levisyin** [comment]
@mattverse I have replaced `slices.Contains` with `osmoutils.Contains`, ci shows that I need to update the changelog too, not sure about this

**Comment by mattverse** [comment]
@levisyin Thanks so much! Can we add changelog so that we can track this PR after merged and include it in our future release?

**Comment by levisyin** [comment]
> @levisyin Thanks so much! Can we add changelog so that we can track this PR after merged and include it in our future release?

@mattverse Changelog has updated~

**Review by mattverse — APPROVED** [review]
LGTM :)

**Pull request merged**